### DISCUSSION TITLE
Saving Heatmap and Segmentation Image in inference mode

### DISCUSSION THREAD

**Discussion by glucasol** [open-edge-platform/anomalib#655] [Q&A]
Hi, I would like to know if it is possible to save the heatmap and the segmentation image in separate images in the inference (torch inference or lightning inference)

**Comment by samet-akcay** [comment]
Hi @glucasol, yes, this is possible with [torch_inference](https://github.com/openvinotoolkit/anomalib/blob/main/tools/inference/torch_inference.py). 

https://github.com/openvinotoolkit/anomalib/blob/e965c8a69259eb5b91c876068233f9b9a65ecf5f/tools/inference/torch_inference.py#L79

where `predictions` is of type [`ImageResult`](https://github.com/openvinotoolkit/anomalib/blob/main/anomalib/post_processing/visualizer.py#L24-L48) that contains the heat map and segmentation as well as other results. By modifying this `torch_inference.py` script, you could save whichever you need.
